python/GafferCortexUI/CompoundPlugValueWidget.py

```diff
@@ -42,8 +42,10 @@
 
 from Qt import QtWidgets
 
-## \deprecated. CompoundPlug itself will be removed in the future, as will this
-# class. Use LayoutPlugValueWidget instead.
+# This was the predecessor to the far superior GafferUI.PlugLayout
+# class that we now use. It survives here as a relic because it is
+# still relied upon by CompoundParameterValueWidget and
+# ClassVectorParameterValueWidget. Do not use it for anything else!
```

Comment:
What is it exactly that is preventing us from using the LayoutPlugValueWidget instead of the CompoundPlugValueWidget?

Reply:
Time really. The Parameter UIs that use CompoundPlugValueWidget are quite intricately tied up with it, and it would take a lot of time to extricate them. CompoundPlugValueWidget had methods you could override to control what children are shown and in what order and with what widgets, and that's used heavily in GafferCortex. The LayoutPlugValueWidget does it all with metadata instead. I suspect that had we had the metadata-driven UI we have now when we wrote GafferCortex, the parameter UIs would be implemented in a completely different way.

Keeping CompoundPlugValueWidget around for private use in GafferCortex is a compromise that lets us get the most important cleanup done without rewriting most of GafferCortexUI and introduce bugs in the process.